taxi/test/test_views.py

```diff
@@ -0,0 +1,127 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Car, Driver
+
+
+class IndexTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        for id_ in range(10):
+            manufacturer = Manufacturer.objects.create(
+                name=f"Brand {id_}",
+                country="Ukraine"
+            )
+
+            Driver.objects.create_user(
+                username=f"driver {id_}",
+                license_number=str(id_)
+            )
+
+            Car.objects.create(
+                model=f"Model {id_}",
+                manufacturer=manufacturer
+            )
+
+    def setUp(self):
+        self.user = get_user_model().objects.create(
+            username="new_driver",
+            password="root1234",
+        )
+        self.client.force_login(self.user)
+
+    def test_index_counters(self):
+        response = self.client.get(reverse("taxi:index"))
+        self.assertEqual(response.context["num_cars"], 10)
+        self.assertEqual(response.context["num_drivers"], 11)
+        self.assertEqual(response.context["num_manufacturers"], 10)
+
+
+class ManufacturerTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        for id_ in range(10):
+            manufacturer = Manufacturer.objects.create(
+                name=f"Brand {id_}",
+                country="Ukraine"
+            )
+
+            Driver.objects.create_user(
+                username=f"driver {id_}",
+                license_number=str(id_)
+            )
+
+            Car.objects.create(
+                model=f"Model {id_}",
+                manufacturer=manufacturer
+            )
```

Comment:
Perhaps it makes sense to move the general code into a separate function and call it in setUpTestData methods?
check, it is the same in ManufacturerTest and IndexTest